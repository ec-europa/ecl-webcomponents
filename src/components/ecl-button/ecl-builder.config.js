const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-button-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-button-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-button-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-button-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
