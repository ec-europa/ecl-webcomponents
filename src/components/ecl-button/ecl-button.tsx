import { Component, Prop, h, Element } from '@stencil/core';

@Component({
  tag: 'ecl-button',
  styleUrls: {
    ec: './build/styles/ecl-button-ec.css',
    eu: './build/styles/ecl-button-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclButton {
  @Element() el: HTMLElement;
  @Prop() styleClass: string = '';
  @Prop() type: string = 'primary';
  @Prop() theme: string = 'ec';

  componentDidRender() {
    if (this.el.getElementsByTagName('ecl-icon')[0] && this.el.querySelector('.ecl-icon')) {
      const slot = this.el.getElementsByTagName('ecl-icon')[0].getAttribute('slot');
      this.el.querySelector('.ecl-icon').classList.add('ecl-button__icon');
      if (slot) {
        this.el.querySelector('.ecl-icon').classList.add(`ecl-button__${slot.substring(0, 5) + '-' + slot.substring(5)}`);
      }
    }
  }

  getClass(): string {
    return [
      `ecl-button`,
      `ecl-button--${this.type}`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <button class={this.getClass()}>
        <span class="ecl-button__container">
          <slot name="icon-before"></slot>
          <span class="ecl-button__label">
            <slot></slot>
          </span>
          <slot name="icon-after"></slot>
        </span>
      </button>
    )
  }
}
