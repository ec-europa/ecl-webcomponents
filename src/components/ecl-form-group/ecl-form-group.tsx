import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-form-group',
  styleUrls: {
    ec: './build/styles/ecl-form-group-ec.css',
    eu: './build/styles/ecl-form-group-eu.css',
  },
  shadow: false,
  scoped: true,
})

export class EclFormGroup {
  @Prop() styleClass: string = '';
  @Prop() theme: string = 'ec';
  @Prop() label: string;
  @Prop() invalid: boolean = false;
  @Prop() required: boolean = false;
  @Prop() requiredText: string = '*';
  @Prop() optionalText: string;
  @Prop() invalidText: string;
  @Prop() helperText: string;
  @Prop() helperId: string;
  @Prop() labelTag: string = 'label';
  @Prop() tag: string = 'div';
  @Prop() name: string;
  @Prop() invalidIcon: string = 'error';
  @Prop() labelClass: string;
  @Prop() ariaLabelOptional: string;
  @Prop() ariaLabelRequired: string;

  getClass(): string {
    return [
      `ecl-form-group`,
      this.styleClass
    ].join(' ');
  }

  render() {
    return (
      <this.tag
        class={this.getClass()}
        aria-describedby={this.helperId}
      >
      { this.label ?
        <this.labelTag
          class={`ecl-form-label ${this.invalid ? 'ecl-form-label--invalid' : ''} ${this.labelClass || ''}`}
          {...(this.ariaLabelRequired ? { 'aria-label': this.ariaLabelRequired } : {})}
        >
          {this.label}
      {this.required && this.requiredText ? (
        <span
          class="ecl-form-label__required" 
          {...(this.ariaLabelRequired ? { 'aria-label': this.ariaLabelRequired } : {})}
        >
          {this.requiredText}
        </span>
      ) : ''}
      { !this.required && this.optionalText ?
        <span
          class="ecl-form-label__optional"
          {...(this.ariaLabelOptional ? { 'aria-label': this.ariaLabelOptional } : {})}
        >
          {this.optionalText}</span> : ''
      }
        </this.labelTag> : ''
      }
      { this.helperText ? 
        <div
          class="ecl-help-block"
        >
          {this.helperText}
        </div> : ''
      }
      { this.invalid && this.invalidText ?
        <div class="ecl-feedback-message">
        { this.invalidIcon ? 
          <ecl-icon
            styleClass={`ecl-feedback-message__icon sc-ecl-form-group-${this.theme}`}
            icon={this.invalidIcon}
            size="m"
          ></ecl-icon> : '' 
        }
          {this.invalidText}
        </div> : '' 
      }
        <slot></slot>
      </this.tag>
    )
  }
}