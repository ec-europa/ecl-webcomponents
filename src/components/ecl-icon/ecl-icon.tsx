import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}