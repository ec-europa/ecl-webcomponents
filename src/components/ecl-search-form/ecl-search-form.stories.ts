export default {
  title: 'Components/form/search-form',
};

const Template = args =>
`<ecl-search-form
    theme="${args.theme}"
    input-id="ecl-search-form"
    width="m"
    placeholder="Placeholder text"
    button-label="Search"
    button-aria-label="Search"
  >
</ecl-search-form>`;

export const SearchForm = Template.bind({});
SearchForm.storyName = 'default';
