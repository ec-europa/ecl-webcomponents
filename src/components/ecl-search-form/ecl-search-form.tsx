import { Component, h, Prop } from '@stencil/core';

@Component({
  tag: 'ecl-search-form',
  styleUrls: {
    ec: './build/styles/ecl-search-form-ec.css',
    eu: './build/styles/ecl-search-form-eu.css',
  },
  shadow: false,
  scoped: true,
})
export class EclSearchForm {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() disabled: boolean = false;
  @Prop() required: boolean = false;
  @Prop() invalid: boolean = false;
  @Prop() invalidText: string;
  @Prop() helperId: string;
  @Prop() helperText: string;
  @Prop() placeholder: string;
  @Prop() inputId: string;
  @Prop() inputDefaultValue: string;
  @Prop() width: string = 'm';
  @Prop() label: string;
  @Prop() type: string;
  @Prop() name: string;
  @Prop() buttonLabel: string;
  @Prop() buttonAriaLabel: string;

  getClass(): string {
    const styleClasses = ['ecl-search-form', this.styleClass];

    return styleClasses.join(' ');
  }

  render() {
    return (
      <form
        class={this.getClass()}
        role="search"
      >
        <ecl-form-group
          style-class={`sc-ecl-search-form-${this.theme}`}
          label={this.label}
          required={this.required}
          invalid={this.invalid}
          invalid-text={this.invalidText}
          theme={this.theme}
        >
          <ecl-input
            style-class={`ecl-search-form__text-input sc-ecl-search-form-${this.theme}`}
            type="search"
            input-class={`ecl-search-form__text-input sc-ecl-search-form-${this.theme}`}
            theme={this.theme}
            input-id={this.inputId}
            required={this.required}
            disabled={this.disabled}
            invalid={this.invalid}
            width="m"
            placeholder={this.placeholder}
            default-value={this.inputDefaultValue}
          >
          </ecl-input>
        </ecl-form-group>
        <ecl-button
          style-class={`ecl-search-form__button sc-ecl-search-form-${this.theme}`}
          aria-label={this.buttonAriaLabel}
          variant="search"
          theme={this.theme}
        >
          {this.buttonLabel}
          <ecl-icon
            style-class={`sc-ecl-search-form-${this.theme}`}
            slot="icon-after"
            icon="search"
            size="xs"
            theme={this.theme}
          >
          </ecl-icon>
        </ecl-button>
      </form>
    );
  }
}