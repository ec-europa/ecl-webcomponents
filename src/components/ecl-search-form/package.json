{
  "name": "@ecl/web-components-search-form",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.13.0",
  "description": "ECL Search form",
  "main": "ecl-search-form.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "@ecl/builder": "3.13.0",
    "@ecl/vanilla-component-search-form": "3.13.0",
    "cross-env": "7.0.3"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
