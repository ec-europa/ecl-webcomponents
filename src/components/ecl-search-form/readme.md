# ecl-search-form



<!-- Auto Generated Below -->


## Properties

| Property          | Attribute           | Description | Type      | Default     |
| ----------------- | ------------------- | ----------- | --------- | ----------- |
| `buttonAriaLabel` | `button-aria-label` |             | `string`  | `undefined` |
| `buttonLabel`     | `button-label`      |             | `string`  | `undefined` |
| `disabled`        | `disabled`          |             | `boolean` | `false`     |
| `helperId`        | `helper-id`         |             | `string`  | `undefined` |
| `helperText`      | `helper-text`       |             | `string`  | `undefined` |
| `inputId`         | `input-id`          |             | `string`  | `undefined` |
| `invalid`         | `invalid`           |             | `boolean` | `false`     |
| `invalidText`     | `invalid-text`      |             | `string`  | `undefined` |
| `label`           | `label`             |             | `string`  | `undefined` |
| `name`            | `name`              |             | `string`  | `undefined` |
| `placeholder`     | `placeholder`       |             | `string`  | `undefined` |
| `required`        | `required`          |             | `boolean` | `false`     |
| `styleClass`      | `style-class`       |             | `string`  | `undefined` |
| `theme`           | `theme`             |             | `string`  | `'ec'`      |
| `type`            | `type`              |             | `string`  | `undefined` |
| `width`           | `width`             |             | `string`  | `'m'`       |


## Dependencies

### Depends on

- [ecl-form-group](../ecl-form-group)
- [ecl-input](../ecl-input)
- [ecl-button](../ecl-button)
- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-search-form --> ecl-form-group
  ecl-search-form --> ecl-input
  ecl-search-form --> ecl-button
  ecl-search-form --> ecl-icon
  ecl-form-group --> ecl-icon
  ecl-input --> ecl-icon
  style ecl-search-form fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
