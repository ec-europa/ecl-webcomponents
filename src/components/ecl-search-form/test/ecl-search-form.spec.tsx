import { newSpecPage } from '@stencil/core/testing';
import { EclSearchForm } from '../ecl-search-form';
import { EclFormGroup } from '../../ecl-form-group/ecl-form-group';
import { EclButton } from '../../ecl-button/ecl-button';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-search-form', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclSearchForm, EclButton, EclIcon, EclFormGroup],
      html: `<ecl-search-form
        theme="ec"
        input-id="ecl-search-form"
        width="m"
        placeholder="Placeholder text"
        button-label="Search"
        button-aria-label="Search"
      >
    </ecl-search-form>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});