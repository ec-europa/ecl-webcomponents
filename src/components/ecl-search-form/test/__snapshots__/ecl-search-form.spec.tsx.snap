// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-search-form renders 1`] = `
<ecl-search-form button-aria-label="Search" button-label="Search" input-id="ecl-search-form" placeholder="Placeholder text" theme="ec" width="m">
  <form class="ecl-search-form" role="search">
    <ecl-form-group style-class="sc-ecl-search-form-ec">
      <!---->
      <fieldset class="ecl-form-group sc-ecl-search-form-ec">
        <ecl-input input-class="ecl-search-form__text-input sc-ecl-search-form-ec" input-id="ecl-search-form" placeholder="Placeholder text" style-class="ecl-search-form__text-input sc-ecl-search-form-ec" theme="ec" type="search" width="m"></ecl-input>
      </fieldset>
    </ecl-form-group>
    <ecl-button aria-label="Search" style-class="ecl-search-form__button sc-ecl-search-form-ec">
      <!---->
      <button class="ecl-button ecl-button--search ecl-search-form__button sc-ecl-search-form-ec" type="submit">
        <span class="ecl-button__container">
          <span class="ecl-button__label">
            Search
          </span>
          <ecl-icon slot="icon-after" style-class="sc-ecl-search-form-ec">
            <svg class="ecl-button__icon ecl-button__icon--after ecl-icon ecl-icon--xs sc-ecl-button-ec sc-ecl-search-form-ec">
              <use xlink:href="/build/images/ec/icons.svg#search"></use>
            </svg>
          </ecl-icon>
        </span>
      </button>
    </ecl-button>
  </form>
</ecl-search-form>
`;
